src/ui/views/ManualDownload.svelte

```diff
@@ -1,23 +1,102 @@
 <script>
   const { ipcRenderer } = require("electron");
+  const path = require('path');
+  const fs = require('fs');
+  
   import {
     manualDownloadFileData,
     manualDownloadGroup,
     eventObject
   } from "../../stores.mjs";
 
-  let downloadedFile;
   let checkingFile = false;
+  let selectedDirectory = "";
+  let constructedFilePath = "";
+  let errorMessage = "";
+
+  // Pick directory and construct full file path
+  async function selectDirectoryAndConstructPath() {
+    errorMessage = ""; // Clear previous errors
+    
+    try {
+      const result = await ipcRenderer.invoke('show-file-dialog', {
+        properties: ['openDirectory'],
+        defaultPath: require('os').homedir() + '/Downloads',
+        title: `Select Directory containing ${$manualDownloadFileData.name}`
+      });
+      
+      if (!result.canceled && result.filePaths && result.filePaths.length > 0) {
+        selectedDirectory = result.filePaths[0];
+        
+        // Sanitize and validate filename
+        const expectedFilename = path.basename($manualDownloadFileData.name);
+        
+        if (!expectedFilename || expectedFilename === '.' || expectedFilename === '..') {
+          errorMessage = "Invalid filename provided";
+          return;
+        }
+        
+        // Join directory path + expected filename
+        constructedFilePath = path.join(selectedDirectory, expectedFilename);
```

Comment:
What does this accomplish? Also this filename is not "provided", it's part of the installer config and the user has no control over that